server/src/schema.js

```diff
@@ -1,31 +1,46 @@
 const {gql} = require('apollo-server');
 
 const typeDefs = gql`
+    scalar Date
+    scalar DateTime
+
     type Query {
+        # Returns certain size of events after the cursor
+        # Reference:
+        #  https://www.apollographql.com/docs/tutorial/resolvers/#paginated-queries 
         events(pageSize: Int, after: String): EventConnection!
+        # Return specific event whose id is 'id'.
+        # If not exist, return null
         event(id: ID!): Event
-
-        # user information about the user currently logged in
+        # Return the user currently logged in
         me: User
+        # Return the user whose id is 'id'.
+        # If not exist, return null
         user(id: ID!): User
     }
 
     type Mutation {
-        authGoogle(accessToken: AuthInput!): AuthResponse
-        authFacebook(accessToken: AuthInput!): AuthResponse
-        logout: Boolean! # success, then True.  
+        signInWithGoogle(googleId: String!): AuthResponse!
+        signUpWithGoogle(
+            googleId: String!
+            email: String!
+            firstName: String!
+            lastName: String!
+            profileImgUrl: String
+        ): AuthResponse!
+        # If successed, then return True.  
```

Comment:
But what happens if logout is unsuccessful? We should log those sessions so we can analyze later. Same for all other unsuccessful responses below.

Reply:
Thank you! I added it to TODO lists.